Final_Project/Main.cpp

```diff
@@ -190,6 +199,7 @@ LRESULT CALLBACK WndProc(HWND hwnd, UINT Message,
 			//generateCtrlFrame(ctrlFrame, 5); //for test; to be removed
 			receiveFrame(ctrlFrame); //for test; to be removed
 			//sendCharacter(hwnd);
+			curState.assign("SEND");
```

Comment:
currently it seems like our curState is not being updated properly. 
I tried using both '=' and string.assign, but both didn't seem to update our static curState variable. 
Let me know if there's a better way to do this.

Reply:
Hm, it seems to be working on my end.. 
I just used `curState = "SEND";` under the IDM_UPLOAD case and when stepping through it updated the variable.

Can you add screenshots or a quick video from your end?

Reply:
you're right it is working . I think it was just an error from my end

Reply:
Actually I'm not sure now... I tried running your branch so that it would stop sending EOT when curState is updated and it just ignores it...

Were you running into that problem?

Reply:
Yea, hmm when i tried again it did look like it was updating curState. 

Similarly in my other branch for refactoring receiving and sending, I ran into it again with updating LAST_EOT_RECEIVED. 
I worked around it by adding an update function in Main.cpp which seemed to work. 
I'll push the updated changed to this same branch to minimize conflicts